bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/handler/EmbyBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+            }
+        }
+    }
+
+    public void sendCommand(String commandURL, String payload) {
+        logger.trace("Sending command: {} with payload: {}", commandURL, payload);
+        if (httputils != null) {
+            try {
+                httputils.doPost(commandURL, payload, 2);
+            } catch (EmbyHttpRetryExceeded e) {
+                logger.debug("The number of retry attempts was exceeded", e.getCause());
+            }
+        }
+    }
+
+    private String getServerAddress() {
+        return config.ipAddress + ":" + config.port;
+    }
+
+    private void establishConnection() {
+        // Ensure scheduler is non-null (BaseBridgeHandler always provides one, but it's annotated @Nullable)
+        final ScheduledExecutorService exec = java.util.Objects.requireNonNull(scheduler, "scheduler must not be null");
+
+        exec.execute(() -> {
+            try {
+                // Build a full HTTP URL from host + port
+                String httpBaseUrl = String.format("http://%s:%d", config.ipAddress, config.port);
+
+                // Call the instance connect(...) with the 5-arg signature
+                connection.connect(httpBaseUrl, config.api, exec, config.refreshInterval, config.bufferSize);
+
+                // Schedule periodic connection checks
+                connectionCheckerFuture = exec.scheduleWithFixedDelay(() -> {
+                    if (!connection.checkConnection()) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Connection could not be established");
+                    }
+                }, CONNECTION_CHECK_INTERVAL_MS, CONNECTION_CHECK_INTERVAL_MS, TimeUnit.MILLISECONDS);
+
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Connection attempt failed: " + e.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void initialize() {
+        // Indicate that we're trying to connect
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Attempting to connect to Emby server");
+
+        // Background setup
+        scheduler.execute(() -> {
+            this.reconnectionCount = 0;
+            this.config = new EmbyBridgeConfiguration();
+            if (callbackIpAddress != null) {
+                logger.debug("The callback ip address is: {}", callbackIpAddress);
+            } else {
+                logger.warn("Callback IP address is not set");
+            }
+            if (config.api != null && !config.api.isEmpty()) {
+                this.httputils = new EmbyHTTPUtils(30, config.api, getServerAddress());
+            }
+            this.connection = new EmbyConnection(this, this.webSocketClient);
+            try {
+                EmbyBridgeConfiguration cfg = checkConfiguration();
```

Comment:
I would prefer to perform a configuration validation at the start of this method. after those checks, set the thingstatus to unknown and start the async task to establish a connection.

Reply:
reordered and added thing status updates